packages/teams-js/src/public/appInstallDialog.ts

```diff
@@ -24,8 +28,19 @@ export namespace appInstallDialog {
       if (!isSupported()) {
         throw new Error('Not supported');
       }
-      sendMessageToParent('appInstallDialog.openAppInstallDialog', [openAPPInstallDialogParams]);
-      resolve();
+      if (runtime.isLegacyTeams) {
```

Comment:
do we need unit tests for these changes?

Reply:
Unit tests have been added, I noticed there are some missing logic gaps in some of the tests (particularly calendar) next steps are to write new tests that will test functionality more extensively, but don't want to hold up this PR with those changes.